frontend/website/src/components/login.svelte

```diff
@@ -0,0 +1,601 @@
+						globalMessage =
+							"An unknown error occured, if the problem persists please contact support";
+						console.error(error);
+					}
+
+					// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+					window.turnstile.reset(document.querySelector("#login-turnstile-container iframe")!.id);
+					return;
+				}
+
+				// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+				token = result.data!.auth.register.token;
+			}
+
+			globalIsError = false;
+			globalMessage = "Success!";
+			sessionToken.set(token);
+			setTimeout(() => {
+				loginMode.set(0);
+			}, 200);
+		}
+
+		await inner();
+
+		loggingIn = false;
+	}
+
+	const turnstileSiteKey = import.meta.env.VITE_CF_TURNSTILE_KEY;
+	function onTurnstileCallback(event: CustomEvent<{ token: string }>) {
+		turnstileToken = event.detail.token;
+	}
+
+	function clearTurnstileToken() {
+		turnstileToken = "";
+	}
+</script>
+
+<svelte:window on:keydown={handleKeyDown} />
+
+<div class="container" class:show on:mousedown={handleOnMouseDown} on:keydown={handleKeyDown}>
+	<div class="popup">
+		<div class="login-title">
+			<h2 class="text-left signup-title">{$loginMode === 1 ? "Login" : "Sign up"}</h2>
+			<h2 class="text-left signup-subtitle">
+				{$loginMode === 1 ? "Don't have an account?" : "Already have an account?"}
+				<span>
+					<button class="link-button" on:click={toggleMode}
+						>{$loginMode === 1 ? "Sign up" : "Sign in"}</button
+					>
+				</span>
+			</h2>
+		</div>
+		<form use:focusTrap={true} on:submit|preventDefault={onSubmit}>
+			<LoginField field={username} />
+			{#if $loginMode === 2}
+				<LoginField field={email} />
+			{/if}
+			<LoginField field={password} />
+			{#if $loginMode === 2}
+				<LoginField field={confirmPassword} />
+			{/if}
+			{#if globalMessage !== ""}
+				<div class="message-holder" class:error={globalIsError}>
+					<span>{globalMessage}</span>
+				</div>
+			{/if}
+			<div id="login-turnstile-container">
+				<Turnstile
+					siteKey={turnstileSiteKey}
+					on:turnstile-callback={onTurnstileCallback}
+					on:turnstile-error={clearTurnstileToken}
+					on:turnstile-expired={clearTurnstileToken}
+					on:turnstile-timeout={clearTurnstileToken}
+				/>
+			</div>
+			<div class="button-group">
+				<input
+					class="button-submit"
+					type="submit"
+					value={loggingIn ? "Loading..." : $loginMode === 1 ? "Login" : "Sign up"}
+					disabled={!formValid || loggingIn}
+				/>
+			</div>
+		</form>
+	</div>
+</div>
+
+<style lang="scss">
+	.container {
+		position: fixed;
+		width: 100vw;
+		height: 100vh;
+		top: 0;
+		left: 0;
+		z-index: 2;
+		background-color: #000000a3;
+		opacity: 0;
+		transition: opacity 0.1s ease-in-out;
+		&.show {
+			transition: opacity 0.2s ease-in-out;
+			opacity: 1;
+		}
+	}
+
+	.link-button {
+		color: #cf634d;
+		font-family: inherit;
+		font-size: inherit;
+		background-color: transparent;
+		margin: 0;
+		padding: 0;
+		outline: 0;
+		border: 0;
+		cursor: pointer;
```

Comment:
We could probably extract cursor pointer for every button into global scss file

Reply:
True, but this can be done at a later stage.